x-pack/platform/plugins/shared/onechat/server/services/tools/esql/utils/to_tool_definition.ts

```diff
@@ -37,10 +37,12 @@ export function toToolDefinition<TSchema extends z.ZodObject<any> = z.ZodObject<
       return {
         results: [
           {
+            toolResultId: getToolResultId(),
             type: ToolResultType.tabularData,
             data: {
-              columns: response.columns,
-              values: response.values,
+              source: 'esql',
+              query: configuration.query,
+              result,
```

Comment:
As per our conversation, the `tabularData` tool now returns both the esql query and result